src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactory.java

```diff
@@ -34,6 +34,15 @@
     BitbucketBuildStatusClient getBuildStatusClient(String revisionSha, BitbucketSCMRepository bitbucketSCMRepo,
                                                     BitbucketCICapabilities ciCapabilities);
 
+    /**
+     * Return a client that can post deployment information to Bitbucket.
+     *
+     * @param revisionSha      the revision for the deployment
+     * @param bitbucketSCMRepo the (Bitbucket) SCM Repo
+     * @return a client that can post deployment information
+     */
+    BitbucketDeploymentClient getDeploymentClient(String revisionSha, BitbucketSCMRepository bitbucketSCMRepo);
```

Comment:
This is the wrong location for this I think. (arguably the build status client is in the wrong place as well). 
The client framework we have is fluent. so your client should sit on the `BitbucketRepositoryClient`. Thus to get hold of the deployment client you go
```java
clientFactory.getProjectClient(projectKey)
.getRepositoryClient(repoSlug)
.getDeploymentClient(revisionSha)
```

Reply:
Unfortunately doing it that way I don't have access to any BitbucketSCMRepository. If I move it to a fluent API, how do you propose we would fit in the `BitbucketSCMRepository`?